frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx

```diff
@@ -580,13 +592,37 @@ function EditAnsibleJobModal(props: {
     ansibleSelection?: string
     ansibleCredentials: ProviderConnection[]
     ansibleTowerTemplateList: string[] | undefined
+    ansibleTowerWorkflowTemplateList: string[] | undefined
     ansibleJob?: ClusterCuratorAnsibleJob
     ansibleJobList?: ClusterCuratorAnsibleJob[]
     setAnsibleJob: (ansibleJob?: ClusterCuratorAnsibleJob, old?: ClusterCuratorAnsibleJob) => void
 }) {
     const { t } = useTranslation()
+    const useStyles = makeStyles({
+        radio: {
+            paddingRight: '10px',
+        },
+    })
+    const classes = useStyles()
     const [ansibleJob, setAnsibleJob] = useState<ClusterCuratorAnsibleJob | undefined>()
+    const [filterForJobTemplates, setFilterForJobTemplates] = useState(true)
     useEffect(() => setAnsibleJob(props.ansibleJob), [props.ansibleJob])
+
+    const newTemplateSelection = (jobName: string | undefined) => {
+        if (ansibleJob) {
+            ansibleJob.type = filterForJobTemplates ? 'Job' : 'Workflow'
```

Comment:
Don't mutate original state. Mutate the copy.

Reply:
Also, does this belong in `newTemplateTypeSelection`? Why are we updating the name in both functions?

Reply:
The name update in `newTemplateTypeSelection` wipes the name field away as we change template types. I guess I don't really need to pass in a jobName for this function, so I will remove that argument.